src/sqlancer/tidb/TiDBProvider.java

```diff
@@ -132,8 +134,19 @@ public void generateDatabase(TiDBGlobalState globalState) throws Exception {
 
     @Override
     public SQLConnection createDatabase(TiDBGlobalState globalState) throws SQLException {
+        host = globalState.getOptions().getHost();
+        port = globalState.getOptions().getPort();
+        if (host == null) {
+            host = "localhost";
+        }
+        if (port == -1) {
+            port = 4000;
+        }
+
         String databaseName = globalState.getDatabaseName();
-        String url = "jdbc:mysql://127.0.0.1:4000/";
+        // String url = "jdbc:mysql://127.0.0.1:4000/";
```

Comment:
Ideally, we could remove the commented-out code (but you don't need to do this for this PR).